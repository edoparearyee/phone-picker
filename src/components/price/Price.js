import React, { Component } from 'react';
import './Price.css';

/**
 * Lists monthly and upfront costs for selected device
 *
 * @class Price
 * @extends {Component}
 */
class Price extends Component {
  render() {
    return (
      <div className="Price">
        {this.props.upfrontPrice ? (
          <div className="Price__upfront">
            <p>
              from{' '}
              <span className="Price__value">£{this.props.upfrontPrice}</span>{' '}
              upfront
            </p>
          </div>
        ) : (
          ''
        )}
        {this.props.monthPrice ? (
          <div className="Price__month">
            <p>
              when you pay{' '}
              <span className="Price__value">£{this.props.monthPrice}</span> a
              month
            </p>
          </div>
        ) : (
          ''
        )}
      </div>
    );
  }
}

export default Price;
