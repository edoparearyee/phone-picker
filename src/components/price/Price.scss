@import '../../styles/functions';
@import '../../styles/variables';

.Price {
  background-color: $light-grey-3;
  display: flex;
  padding: rem(20px);
}

.Price__upfront,
.Price__month {
  padding: rem(20px);
  flex: 1 0 50%;
  max-width: 50%;
}

.Price__upfront {
  border-right: rem(1px) solid $light-grey-1;
}

.Price__value {
  color: $red;
  font-size: rem(32px);
  font-weight: 700;
}
