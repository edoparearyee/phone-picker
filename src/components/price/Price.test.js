import React from 'react';
import ReactDOM from 'react-dom';
import Price from './Price';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Price />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('Price displays month and upfront prices', () => {
  const component = renderer.create(
    <Price monthPrice={43.2} fullPrice={1149} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
