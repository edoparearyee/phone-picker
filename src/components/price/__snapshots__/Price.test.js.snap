// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price displays month and upfront prices 1`] = `
<div
  className="Price"
>
  
  <div
    className="Price__month"
  >
    <p>
      when you pay
       
      <span
        className="Price__value"
      >
        Â£
        43.2
      </span>
       a month
    </p>
  </div>
</div>
`;
